src/SplitProvider.tsx

```diff
@@ -33,74 +25,75 @@ export const SplitContext: SplitReactContext = createContext<
  *    <App />
  *  </SplitProvider>
  */
-const SplitProvider = class extends React.Component<
-  ISplitProviderProps,
-  ISplitContextValues
-> {
-  constructor(props: ISplitProviderProps) {
-    super(props);
+const SplitProvider = ({ config, children }: ISplitProviderProps) => {
+  const {
+    startup = {},
+    scheduler = {},
+    core,
+    features = {},
+    storage = {},
+    debug,
+    impressionListener,
+  } = config;
+  const clientDeps = [
```

Comment:
To re-create the client when config changes and to _not_ re-create the client when config _doesn't_ change.

`useEffect` uses this list as a way to memoize the effect called. So if the dependency list is the same as it was the last time the function was ran, it doesn't call the function again.

It does identity checks on each value in the list, which unfortunate since config is an object.
Take this for example:
```tsx
<SplitProvider
  config={{
    core: {
      key: 'my key'
    }
  }}
>
  ...
</SplitProvider>
```
Every time the component including ^ that re-renders `SplitProvider` is passed a _new_ object for its config property.
If our create client logic looked like this
```tsx
useEffect(() => {
  // create client
}, [config]);
```
Then the effect would be called every render as the config object's identity would be different.

So we get around this by checking all the values in the config object as scalars instead of object itself. It sucks, but I don't think we should put it on the consumers to memoize their config objects.

Open to suggestions about different ways to do this too. I wonder if there is a more performant way.

Reply:
We could use `JSON.stringify` to do a deep-equality check.

Reply:
> also more secure, let's say another prop is injected in the split config object, we would need to update our code to reflect that.

Right, exactly why I think this is hacky.

`JSON.stringify` also seems hacky to me too though. What if the key order changes and then the comparison string is different? I would be more comfortable with lodash.isEqual...

Reply:
I just don't want to add any other libraries, I think `JSON.stringify` is a great option, I don't think it's hacky, `isEqual` is not even available in recent versions, now it's `isEqualWith`. I'm not quite sure I understood "what if the key order change", from the user perspective or from split.io sdk perspective you mean?

Reply:
Yeah I hear you. It adds complexity. Where do you see `isEqual` not available?

`JSON.stringify` will fail like this:
```ts
'{foo: "red", bar: "blue"}' === '{bar: "blue", foo: "red"}'
```
If we just want to note this as a warning and move on I guess that's fine.

That also wouldn't work for the config options that are functions, like `impressionListener`.

Reply:
I was reading the lodash source code (https://github.com/lodash/lodash), I couldn't find it there, I think I'm wrong the documentation still has it.

I understand that order will change the results of `JSON.stringify` I guess my question is, in which context the order of split.io config object might change, once it's setup I doubt users will keep changing its order. The function is irrelevant I think, we're just trying to do a deep equality check.

But I think either your solution or `JSON.stringify` is fine, I'll let you decide.

Reply:
I think lodash is a better solution but I'm hesitant to go with it because of what it could imply with compilation. I'm really weak with what's needed to compile and publish TS/JS libraries.

We can always optimize down the road.

Reply:
I agree, are you going to leave it as it is or use `JSON.stringify`?

Reply:
I'll change it to `JSON.stringify`. Maybe I can wrap it in a memo call so we don't incur the `JSON.stringify` cost if the object identity hasn't changed.